### PR TITLE
refactor: update display, quota guard and usage report logic for enterprise users

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Update console display and core quota guard/usage report logic for customized enterprise plans, with the following changes:
1. [console] update the plan name display on tenant settings page -> Plan and billing section
2. [console] update the plan name display on tenant selector dropdown items
3. [console] do not show add-on tag and paywall feature tag for enterprise plan
4. [core/quota guard] do not block enterprise plan add-on related items when checking quota guard upon adding resources
5. [core/usage report] also report usage status for enterprise plans

4/5 enable enterprise plan users to breakthrough add-on related limit in some edge cases to maintain a good user experience.

## Testing

Tested locally, the display and @logto/cloud response all looks good.
image
image
image
image

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/console/src/components/PlanUsage/PlanUsageCard/index.module.scss [hunk 1]
```diff
@@ -31,6 +31,10 @@
 .usageTip {
   font: var(--font-body-2);
   color: var(--color-text-secondary);
+
+  &.hideForEnterprisePlan {
```
Comment (charIeszhao) [inline packages/console/src/components/PlanUsage/PlanUsageCard/index.module.scss:35]: Should just be `&.hidden`

File: packages/console/src/components/PlanUsage/PlanUsageCard/index.tsx [hunk 2]
```diff
@@ -32,6 +33,7 @@ function PlanUsageCard({
   usageKey,
   titleKey,
   tooltipKey,
+  isUsageTipHidden,
```
Comment (charIeszhao) [inline packages/console/src/components/PlanUsage/PlanUsageCard/index.tsx:36]: Can we use `isEnterprisePlan` inside this component? So you don't need to pass the param

File: packages/console/src/pages/ApiResources/components/CreateForm/Footer.tsx [hunk 3]
```diff
@@ -24,7 +24,7 @@ type Props = {
 function Footer({ isCreationLoading, onClickCreate }: Props) {
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
   const {
-    currentSubscription: { planId, isAddOnAvailable },
+    currentSubscription: { planId, isAddOnAvailable, isEnterprisePlan },
```
Comment (charIeszhao) [inline packages/console/src/pages/ApiResources/components/CreateForm/Footer.tsx:27]: Can we add a new variable in the context, like `isProOrEnterprisePlan`? This way, you don't need to write `planId === ReservedPlanId.Pro || isEnterprisePlan` everywhere

**Review by simeng-li — APPROVED** [review]
core updated LGTM

File: packages/console/src/pages/TenantSettings/BillingHistory/index.tsx [hunk 4]
```diff
@@ -50,7 +51,19 @@ function BillingHistory() {
...
             title: <DynamicT forKey="subscription.billing_history.invoice_column" />,
             dataIndex: 'planName',
-            render: ({ skuId, periodStart, periodEnd }) => {
+            render: ({ skuId: rawSkuId, periodStart, periodEnd }) => {
+              /**
+               * @remarks
+               * The `skuId` should be either ReservedPlanId.Dev, ReservedPlanId.Pro, ReservedPlanId.Admin, ReservedPlanId.Free, or a random string.
+               * Except for the random string, which corresponds to the custom enterprise plan, other `skuId` values correspond to specific Reserved Plans.
+               */
+              const skuId = conditional(
```
Comment (charIeszhao) [inline packages/console/src/pages/TenantSettings/BillingHistory/index.tsx:60]: Is this `conditional` needed?

File: packages/console/src/pages/ApiResources/components/CreateForm/Footer.tsx (hunk 3 above)
Comment (darcyYe) [reply packages/console/src/pages/ApiResources/components/CreateForm/Footer.tsx:27]: For now, we can define a function here in console.

File: packages/console/src/components/PlanUsage/PlanUsageCard/index.tsx (hunk 2 above)
Comment (darcyYe) [reply packages/console/src/components/PlanUsage/PlanUsageCard/index.tsx:36]: Was using `isEnterprisePlan` flag in the component, but since we will have different display logic for all quota usages, better to keep the flag simple and we can configure them from outer scope.

**Review by charIeszhao — APPROVED** [review]
